src/vendedores.ts

```diff
@@ -1,31 +1,159 @@
-import { any } from "ramda";
+import { any, uniq } from "ramda";
 
 abstract class Vendedor {
+  constructor(public certificaciones: Certificacion[]) {}
+
   abstract puedeTrabajarEn(ciudad: Ciudad): boolean;
+  abstract esInfluyente(): boolean;
+
+  esVersatil(): boolean {
+    const alMenosUnaSobreProductos = any(
+      (cert) => cert.esSobreProductos,
+      this.certificaciones
+    );
+    const alMenosUnaNoSobreProductos = any(
+      (cert) => !cert.esSobreProductos,
+      this.certificaciones
+    );
+
+    if (
+      this.certificaciones.length >= 3 &&
+      alMenosUnaSobreProductos &&
+      alMenosUnaNoSobreProductos
+    ) {
+      return true;
+    }
+    return false;
+  }
+
+  esFirme(): boolean {
+    if (this.puntaje() >= 30) return true;
+    return false;
+  }
```

Comment:
Cuando usás algún operador que devuelve booleano (`>`, `<=`, `&&`, etc), no tiene sentido además hacer un `if`, porque lo que te devuelve ya es booleano.

Te corrijo este solo, pero vale lo mismo para todos los lugares donde pusiste `if (...) { return true } return false`.

```suggestion
  esFirme(): boolean {
    return this.puntaje() >= 30;
  }
```